test/dataDefaulter.js

```diff
@@ -0,0 +1,143 @@
+        street: "50 Ferncroft Rd",
+        city: "Danvers",
+        zip: "01923",
+        signUpLink: "https://curative.com/sites/24181",
+        hasAvailability: true,
+        availability: {
+            "02/27/2021": {
+                numberAvailableAppointments: 1,
+                hasAvailability: true,
+            },
+            "02/28/2021": {
+                numberAvailableAppointments: 0,
+                hasAvailability: false,
+            },
+        },
+    },
+];
+
+const realisticDefaultData = [
+    {
+        id: 24181,
+        name: "DoubleTree Hotel - Danvers",
+        street: "50 Ferncroft Rd",
+        city: "Danvers",
+        zip: "01923",
+        signUpLink: "https://curative.com/sites/24181",
+        hasAvailability: false,
+        availability: {
+            "02/18/2021": {
+                numberAvailableAppointments: 0,
+                hasAvailability: false,
+            },
+            "02/19/2021": {
+                numberAvailableAppointments: 0,
+                hasAvailability: false,
+            },
+        },
+    },
+    {
+        name: "Family Practice Group",
+        street: "11 Water Street, Suite 1-A",
+        city: "Arlington",
+        zip: "02476",
+        website: "https://bookfpg.timetap.com/#/",
+        signUpLink: "https://bookfpg.timetap.com/#/",
+        hasAvailability: false,
+    },
+];
+
+describe("Simple behavior", () => {
+    it("should return scraped results when no cache values are present", () => {
+        assert.deepStrictEqual(
+            dataDefaulter.addDefaultsToResults(realisticTestData, []),
+            realisticTestData
+        );
+    });
+    it("should return cached results when no scraped results are present", () => {
+        assert.deepStrictEqual(
+            dataDefaulter.addDefaultsToResults([], realisticDefaultData),
+            realisticDefaultData
+        );
+    });
+});
+
+describe("Conditionally inserting defaults", () => {
+    const finalResults = dataDefaulter.addDefaultsToResults(
+        realisticTestData,
+        realisticDefaultData
+    );
+    it("has the correct number of entries", () => {
+        // Atrius, Danvers, and then default Family Practice Group
+        assert.strictEqual(finalResults.length, 3);
+    });
+    it("does not replace data with cache", () => {
+        // Danvers stays the same
+        assert.deepStrictEqual(finalResults[1], realisticTestData[1]);
+    });
+
+    it("adds default data where an entry doesn't exist", () => {
+        // Family Practice Group is added to the results array
+        assert.deepStrictEqual(finalResults[2], realisticDefaultData[1]);
+    });
+});
+
+describe("Tolerance for stale data", () => {
+    const secondsOfTolerance = 60;
+    it("does not include stale data if specified", () => {
+        const timestampedDefault = {
+            ...realisticDefaultData[0],
+            timestamp: new Date() - (secondsOfTolerance + 1) * 1000,
```

Comment:
Again, unspecified date arithmetic. 
I'm not sure what the `+ 1` is doing there.

Reply:
making sure the timestamp is before the threshold (1 second older). I don't see anything saying subtraction isn't supported with Date().